### ISSUE TITLE
!TupleReturn for non-destructuring assignments

### ISSUE THREAD

**Issue by apemanzilla** [Perryvw/TypescriptToLua#110]
Right now, using `!TupleReturn` only affects destructuring assignments. For example, the following code...

```typescript
/** !TupleReturn */
declare function something(): string[]

const got = something();
```

... should generate

```lua
local got = { something() } -- or table.pack if you prefer
```

rather than 

```lua
local got = something() -- only the first value is captured, and not in a table as expected
```

**Comment by Perryvw** [comment]
Actually only returning the first variable IS expected behavior, if you execute the following lua:
```lua
function myFunc() return 3, 4 end
local a = myFunc()
print(a)
```
You will see it only prints 3. I don't want to limit this coding pattern. If you want all values you should probably explicitly write
```ts
const got = [something()];
```

I will close this issue unless you have a good reason why this needs to be implemented as you suggested.

**Comment by apemanzilla** [comment]
Explicitly wrapping the return in an array messes with the type system. If you do `let x = something();`, `x` is typed as `string[]` rather than the real type `string`. Fixing this without wrapping all non-desttucturing `!TupleReturns` would require changing the type checker itself as far as I know, and IMO also makes it less clear to users how things work.

Another issue is when you're not assigning something at all, but passing to another call directly. This again breaks the type checker, and also can have side effects where arguments are shifted down the list by those returned. If you use `print(something(), "hello")` it will first print the entire set of values returned by `something()`, rather than only the first value.

My proposal is to always wrap `!TupleReturn` calls with a table, except for when using a destructing assignment to an array. This means the expected types are accurate, and prevents unexpected side effects when not assigning to a variable.

**Comment by Perryvw** [comment]
The type checker argument won me over, it is indeed strange to have something like
```ts
/** !TupleReturn */
declare function tupleFunction(): [number, string];
const val = tupleFunction();
```
Since the TS type checker will thing val is a tuple, while actually it is a number.

Implemented this in e4b718b5bf81fd5d55d5fc1ee11f397e206effe5

**Comment by apemanzilla** [comment]
Thank you!
